beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/cleaner/MetadataCleanup.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.metadata.cleanup.cleaner;
+
+import java.time.LocalDateTime;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+
+public interface MetadataCleanup {
+
+  Page<HousekeepingMetadata> findRecordsToClean(LocalDateTime instant, Pageable pageable);
+
+  void cleanupContent(HousekeepingMetadata housekeepingMetadata, LocalDateTime instant, boolean dryRunEnabled);
+
+  boolean cleanupMetadata(HousekeepingMetadata housekeepingMetadata, LocalDateTime instant, boolean dryRunEnabled);
```

Comment:
why you need this? it's exactly the same as cleanupContent except for the method name?

Reply:
https://github.com/ExpediaGroup/beekeeper/blob/91fbd50447/beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/MetadataHandler.java#L51

Its for when theres a dry run. The idea is that `cleanupContent` will call `cleanupMetadata` and then `updateAttemptsAndStatus` - we dont want attempts and status to be updated in a dry run so only cleanup metadata is called instead. 
Since refactoring though its quite unclear, so I'll change this.